### PR TITLE
Change scheduler metrics to conform metrics guidelines

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/sig instrumentation

**What this PR does / why we need it**:

As part of [kubernetes metrics overhaul](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/0031-kubernetes-metrics-overhaul.md), change scheduler metrics to conform [Kubernetes metrics instrumentation guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/instrumentation.md).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This patch does not remove the existing metrics but mark them as deprecated.
We need 2 releases for users to convert monitoring configuration.

**Does this PR introduce a user-facing change?**:

```release-note
Change scheduler metrics to conform metrics guidelines.
The following metrics are deprecated, and will be removed in a future release:
* `e2e_scheduling_latency_microseconds`
* `scheduling_algorithm_latency_microseconds`
* `scheduling_algorithm_predicate_evaluation`
* `scheduling_algorithm_priority_evaluation`
* `scheduling_algorithm_preemption_evaluation`
* `binding_latency_microseconds`
Please convert to the following metrics:
* `e2e_scheduling_latency_seconds`
* `scheduling_algorithm_latency_seconds`
* `scheduling_algorithm_predicate_evaluation_seconds`
* `scheduling_algorithm_priority_evaluation_seconds`
* `scheduling_algorithm_preemption_evaluation_seconds`
* `binding_latency_seconds`
```

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -182,14 +182,16 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 			FailedPredicates: failedPredicateMap,
 		}
 	}
-	metrics.SchedulingAlgorithmPredicateEvaluationDuration.Observe(metrics.SinceInMicroseconds(startPredicateEvalTime))
+	metrics.SchedulingAlgorithmPredicateEvaluationDuration.Observe(metrics.SinceInSeconds(startPredicateEvalTime))
```
Comment (Huang-Wei) [inline pkg/scheduler/core/generic_scheduler.go]: Are you saying replacing "microseconds" with "seconds" is a common pattern? If yes, can you show that where it's documented?

Comment (danielqsj) [reply pkg/scheduler/core/generic_scheduler.go]: Correct.
From the KEP [kubernetes-metrics-overhaul](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/0031-kubernetes-metrics-overhaul.md), Kubernetes metrics should follow Prometheus best practices.
The `seconds` is the [suggested base unit](https://prometheus.io/docs/practices/naming/#base-units) for time type metrics. And `seconds` already been widely used in other Kubernetes metrics.
So we should change these metrics to let them more consistent with others.

**Review by Huang-Wei — COMMENTED** [review]
A nit: please add explicit "DEPRECATED" notice (maybe in release-note section) to notify users that original "microseconds" metrics will be deprecated, so that we can remove them in future releases.

Otherwise
/lgtm

**Comment by danielqsj** [comment]
@Huang-Wei thanks, release-note is updated.

File: pkg/scheduler/metrics/metrics.go [hunk 2]
```diff
@@ -71,50 +71,98 @@ var (
...
+			Help:      "E2e scheduling latency in seconds (scheduling algorithm + binding)",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 15),
+		},
+	)
+	DeprecatedE2eSchedulingLatency = prometheus.NewHistogram(
 		prometheus.HistogramOpts{
 			Subsystem: SchedulerSubsystem,
 			Name:      "e2e_scheduling_latency_microseconds",
-			Help:      "E2e scheduling latency (scheduling algorithm + binding)",
+			Help:      "E2e scheduling latency in microseconds (scheduling algorithm + binding)",
```
Comment (ravisantoshgudimetla) [inline pkg/scheduler/metrics/metrics.go]: I think adding `deprecated` in help might be useful.

**Comment by brancz** [comment]
Agreed with @ravisantoshgudimetla could you start each of the deprecated metric's descriptions with "(deprecated)"? Thanks! Otherwise this lgtm!

**Comment by danielqsj** [comment]
@brancz @Huang-Wei rebased

**Comment by brancz** [comment]
cc @kubernetes/sig-scheduling-pr-reviews

**Review by ravisantoshgudimetla — APPROVED** [review]
Thank you @danielqsj for working on this PR.

/lgtm

**Pull request merged**